src/mainHeader.ts

```diff
@@ -1,14 +1,52 @@
 /* eslint-disable no-control-regex */
+
+import { IOBuffer } from 'iobuffer';
+
 import { xzwTypes, yTypes, experimentSettings } from './types';
-import { getFlagParameters, longToDate } from './utility';
+import { getFlagParameters, longToDate, FlagParameters } from './utility';
+
+export interface Header {
+  fileVersion: number;
+  parameters: FlagParameters;
+  experimentType: string;
+  exponentY: number;
+  numberPoints: number;
+  startingX: number;
+  endingX: number;
+  spectra: number;
+  xUnitsType: string | number;
+  yUnitsType: string;
+  zUnitsType: string;
+  postingDisposition: number;
+  date: string;
+  resolutionDescription: string;
+  sourceInstrumentDescription: string;
+  peakPointNumber: number;
+  spare: number[];
+  memo: string;
+  xyzLabels: string;
+  logOffset: number;
+  modifiedFlag: number;
+  processingCode: number;
+  calibrationLevel: number;
+  subMethodSampleInjectionNumber: number;
+  concentrationFactor: number;
+  methodFile: string;
+  zSubIncrement: number;
+  wPlanes: number;
+  wPlaneIncrement: number;
+  wAxisUnits: string;
+  reserved: string;
+  scans?: number;
+}
 
 /**
  * Main header parsing - First 512/256 bytes (new/old format)
- * @param {object} buffer SPC buffer
- * @return {object} Main header
+ * @param {IOBuffer} buffer SPC buffer
```

Comment:
If the file is '.ts' we should remove the type definition from jsdoc.
This is redundant with the type definition that is present in the function call.

Reply:
Thank you for your comment. I made the necessary changes in the last commit